### PR TITLE
Custom tokens account

### PR DESCRIPTION
**Pull request by rowandh**

### CODE REVIEW

File: StratisCore.UI/src/app/wallet/tokens/components/tokens.component.ts [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    private clipboardService: ClipboardService,
+    private genericModalService: ModalService,
+    private modalService: NgbModal,
+    private globalService: GlobalService,
+    private currentAccountService: CurrentAccountService) {
+
+    this.walletName = this.globalService.getWalletName();
+    
+    this.availableTokens = this.tokenService.GetAvailableTokens();
+    this.availableTokens.push(new Token('Custom', 'custom', 'custom'));
```
Comment (rowandh) [inline StratisCore.UI/src/app/wallet/tokens/components/tokens.component.ts:69]: Remove

File: StratisCore.UI/src/app/wallet/tokens/components/add-token/add-token.component.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    const tickerCall = new LocalCallRequest(address, address, 'Symbol');
+
+    this.loading = true;
+
+    // Add the token if valid token contract exists
+    this.tokenService
+      .LocalCall(tickerCall)
+      .pipe(
+        take(1),
+        takeUntil(this.disposed$),
```
Comment (rowandh) [inline StratisCore.UI/src/app/wallet/tokens/components/add-token/add-token.component.ts:81]: Remove takeUntil

File: StratisCore.UI/src/app/wallet/tokens/components/tokens.component.ts [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+  styleUrls: ['./tokens.component.css']
+})
+@Mixin([Disposable])
+export class TokensComponent implements OnInit, OnDestroy, Disposable {
+  balance: number;
+  coinUnit: string;
+  tokenBalanceRefreshRequested$ = new Subject<SavedToken[]>();
+  addresses: string[];
+  disposed$ = new ReplaySubject<boolean>();
+  dispose: () => void;
```
Comment (rowandh) [inline StratisCore.UI/src/app/wallet/tokens/components/tokens.component.ts]: This should call `disposed$.next()` and `disposed$.completed()`

File: StratisCore.UI/src/app/wallet/dashboard/dashboard.component.ts [hunk 4]
```diff
@@ -112,11 +113,32 @@ export class DashboardComponent implements OnInit, OnDestroy {
...
+  private setWalletBalance(balanceResponse: any) {
+    if (!this.accountsEnabled) {
+      // TO DO - add account feature instead of using first entry in array
+      this.confirmedBalance = balanceResponse.balances[0].amountConfirmed;
+      this.unconfirmedBalance = balanceResponse.balances[0].amountUnconfirmed;
+      this.spendableBalance = balanceResponse.balances[0].spendableAmount;
+    } else {
+      this.confirmedBalance = balanceResponse.amountConfirmed;
+      this.unconfirmedBalance = balanceResponse.amountUnconfirmed;
+      this.spendableBalance = balanceResponse.spendableAmount;
```
Comment (ianadavies) [inline StratisCore.UI/src/app/wallet/dashboard/dashboard.component.ts:125]: `balanceResponse.spendableAmount` does not exist on the response from the WalletController it returns an AddressBalance model see below : or am I missing something?

```
public class AddressBalanceModel
    {
        [JsonProperty(PropertyName = "address")]
        public string Address { get; set; }

        [JsonProperty(PropertyName = "coinType")]
        public CoinType CoinType { get; set; }

        [JsonProperty(PropertyName = "amountConfirmed")]
        public Money AmountConfirmed { get; set; }

        [JsonProperty(PropertyName = "amountUnconfirmed")]
        public Money AmountUnconfirmed { get; set; }
    }
}
```

**Pull request merged**